x-pack/plugins/observability_solution/inventory/public/hooks/use_unified_search_url.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { FilterStateStore } from '@kbn/es-query';
+import { useUrlState } from '@kbn/observability-shared-plugin/public';
+import { enumeration } from '@kbn/securitysolution-io-ts-types';
+import { fold } from 'fp-ts/lib/Either';
+import { constant, identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as t from 'io-ts';
+import { useReducer } from 'react';
+import deepEqual from 'fast-deep-equal';
+
+const FilterRT = t.intersection([
+  t.type({
+    meta: t.partial({
+      alias: t.union([t.null, t.string]),
+      disabled: t.boolean,
+      negate: t.boolean,
+      controlledBy: t.string,
+      group: t.string,
+      index: t.string,
+      isMultiIndex: t.boolean,
+      type: t.string,
+      key: t.string,
+      params: t.any,
+      value: t.any,
+    }),
+  }),
+  t.partial({
+    query: t.record(t.string, t.any),
+    $state: t.type({
+      store: enumeration('FilterStateStore', FilterStateStore),
+    }),
+  }),
+]);
+const FiltersRT = t.array(FilterRT);
+
+const QueryStateRT = t.type({
+  language: t.string,
+  query: t.union([t.string, t.record(t.string, t.any)]),
+});
```

Comment:
I wonder if these should also be centralized somewhere.

Reply:
Ideally, this should be provided by Unified Search plugin. I thought about moving this hook to a common place but in the end, I would overcomplicate it. So I kept it here too.